File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/api/GithubService.java
package ru.avystavkin.githubgists.api;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import ru.avystavkin.githubgists.models.server.GistCommitServer;
import ru.avystavkin.githubgists.models.server.GistServer;

public interface GithubService {

    @GET("/gists/public")
    Observable<List<GistServer>> gists();

    @GET("/gists/{gist_id}")
    Observable<GistServer> gist_detail(@Path("gist_id") String id);

    @GET("/gists/{gist_id}/commits")
    Observable<List<GistCommitServer>> gist_commits(@Path("gist_id") String id);

    @GET("/users/{name}/gists")
    Observable<List<GistServer>>  user_detail(@Path("name") String name);

}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/api/GithubService.kt
package ru.avystavkin.githubgists.api

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import ru.avystavkin.githubgists.models.server.GistCommitServer
import ru.avystavkin.githubgists.models.server.GistServer

interface GithubService {

    @GET("/gists/public")
    fun gists(): Observable<List<GistServer>>

    @GET("/gists/{gist_id}")
    fun gist_detail(@Path("gist_id") id: String): Observable<GistServer>

    @GET("/gists/{gist_id}/commits")
    fun gist_commits(@Path("gist_id") id: String): Observable<List<GistCommitServer>>

    @GET("/users/{name}/gists")
    fun user_detail(@Path("name") name: String): Observable<List<GistServer>>

}
